cmd/admin-rpc-client.go

```diff
@@ -146,9 +167,58 @@ func sendServiceCmd(cps adminPeers, cmd serviceSignal) {
 		wg.Add(1)
 		go func(idx int) {
 			defer wg.Done()
-			errs[idx] = invokeServiceCmd(remotePeers[idx], cmd)
+			// we use idx+1 because remotePeers slice is 1 position shifted w.r.t cps
+			errs[idx+1] = invokeServiceCmd(remotePeers[idx], cmd)
 		}(i)
 	}
 	wg.Wait()
 	errs[0] = invokeServiceCmd(cps[0], cmd)
 }
+
+func fetchLocks(cps adminPeers, bucket, prefix string, relTime time.Duration) ([]VolumeLockInfo, error) {
+	// Used to aggregate volume lock information from all nodes.
+	allLocks := make([][]VolumeLockInfo, len(cps))
+	errs := make([]error, len(cps))
+	var wg sync.WaitGroup
+	remotePeers := cps[1:]
+	for i := range remotePeers {
+		wg.Add(1)
+		go func(idx int) {
+			defer wg.Done()
+			// `remotePeers` is right-shifted by one position relative to `cps`
+			relIdx := idx + 1
+			allLocks[relIdx], errs[relIdx] = cps[relIdx].cmdRunner.ListLocks(bucket, prefix, relTime)
+		}(i)
+	}
+	wg.Wait()
+	allLocks[0], errs[0] = cps[0].cmdRunner.ListLocks(bucket, prefix, relTime)
+
+	// Summarizing errors received for ListLocks RPC across all
+	// nodes.  N B the possible unavailability of quorum in errors
+	// applies only to distributed setup.
+	errCount, err := reduceErrs(errs, []error{})
+	if err != nil {
+		if errCount >= (len(cps)/2 + 1) {
+			return nil, err
+		}
+		return nil, errXLWriteQuorum
```

Comment:
I am wondering if need to do this, how important are these errors for the user..From listing point of view they are not that interesting, just logging on the server might be good enough. Only in case of clear locks we might have to send an error.